test-requirements.txt

```diff
@@ -1,4 +1,4 @@
-juju==3.0.4
+juju==3.1.2.0
```

Comment:
Should we pin the version?

Reply:
yes :)
according to the [release notes of `python-libjuju`](https://github.com/juju/python-libjuju/releases).
`3.1.2.0` `This release works with any Juju 3.x controller.`